astroid/nodes/node_classes.py

```diff
@@ -2061,7 +2061,14 @@ def __init__(
         :param end_col_offset: The end column this node appears on in the
             source code. Note: This is after the last symbol.
         """
-        self.value: Any = value
+        if getattr(value, "__name__", None) == "__doc__":
+            raise AssertionError(  # pragma: no cover
```

Comment:
I don't think this belongs in this PR right?

Reply:
The first three commits in this PR will be merged in the previous PRs. Unfortunately, they are a necessity. The issue that these commits fix are uncovered by the last commit, so without them there are going to be errors.